pkg/ebpftracer/policy.go

```diff
@@ -14,8 +15,9 @@ type Policy struct {
 	Output          PolicyOutputConfig
 }
 
-// PreEventFilter allows for filtering of events coming from the kernel before they are decoded
-type PreEventFilter func(ctx *types.EventContext) error
+// PreEventFilter allows for filtering of events coming from the kernel before they are decoded.
+// Parsed args should be returned if filter passes.
+type PreEventFilter func(ctx *types.EventContext, decoder *decoder.Decoder) (types.Args, error)
```

Comment:
I do not think this is a good idea. This means that we can have a single pre filter max, as otherwise it would mean to decode the args over and over again.

Reply:
I can add validation to have only one pre filter per event type.